File: python/print/print.py
#!/usr/bin/env python
# -*- coding=utf8 -*-
"""
# Author: wanghan
# Created Time : 2018年07月23日 14:01:29
# File Name: print.py
# Description:
"""

import os
import sys
import time
#from __future__ import print_function

if __name__ == '__main__':
    #print('hello',end='') # python 3.x
    print'Hello World!',
    sys.stdout.flush()
    time.sleep(1)
    print('\rProcess     ')
    time.sleep(1)
    sys.stdout.flush()
    while True :
        print '\r■■■■■■■■■■ 100%',
        time.sleep(1)
        sys.stdout.flush()
        print '\r■■■■■■■■■  90%',
        time.sleep(1)
        sys.stdout.flush()
        print '\r■■■■■■■■   80%',
        time.sleep(1)
        sys.stdout.flush()
        print '\r■■■■■■■    70%',
        time.sleep(1)
        sys.stdout.flush()
        print '\r■■■■■■     60%',
        time.sleep(1)
        sys.stdout.flush()
        print '\r■■■■■      50%',
        time.sleep(1)
        sys.stdout.flush()
        print '\r■■■■       40%',
        time.sleep(1)
        sys.stdout.flush()
        print '\r■■■        30%',
        time.sleep(1)
        sys.stdout.flush()
        print '\r■■         20%',
        time.sleep(1)
        sys.stdout.flush()
        print '\r■          10%',
        time.sleep(1)
        sys.stdout.flush()
        print '\r            0%',
        time.sleep(1)
        sys.stdout.flush()

File: c_test/print.c
/*************************************************************************
 @Author: wanghan
 @Created Time : Mon 23 Jul 2018 02:39:16 PM CST
 @File Name: print.c
 @Description:
 ************************************************************************/
#include <stdio.h>
#include <unistd.h>

int main(int argc, const char *argv[])
{
    int i;
    char string_line[1024] = {};
    for(i=0;i<=100;i++){
        sprintf(string_line,"\r%02d percent",i);
        printf(string_line);
        fflush(stdout);
        sleep(1);
    }
    return 0;
}
